src/main/scala/com/amazon/deequ/analyzers/KLLSketch.scala

```diff
@@ -0,0 +1,122 @@
+package com.amazon.deequ.analyzers
+
+import com.amazon.deequ.analyzers.catalyst.KLLSketchSerializer
+import com.amazon.deequ.analyzers.runners.IllegalAnalyzerParameterException
+import com.amazon.deequ.metrics.{BucketDistribution, BucketValue, KLLMetric}
+import scala.collection.mutable.ListBuffer
+import scala.util.Try
+import java.nio.ByteBuffer
+import com.amazon.deequ.analyzers.Analyzers._
+import com.amazon.deequ.analyzers.Preconditions.{hasColumn, isNumeric}
+import com.amazon.deequ.analyzers.runners.MetricCalculationException
+import org.apache.spark.sql.DeequFunctions.stateful_kll
+import org.apache.spark.sql.types.StructType
+import org.apache.spark.sql.{Column, Row}
+
+import scala.util.Failure
+
+/** State for KLL Sketches */
+case class KLLState(qSketch: QuantileNonSample[Double], globalMax: Double, globalMin: Double)
```

Comment:
Please format according to our [coding conventions](https://github.com/databricks/scala-style-guide)

Reply:
Yes I have read it and try to format my code according to conventions. But I could miss out/ misunderstand something, therefore please take a second look after I make the next commit.